docker/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM ubuntu:rolling
+
```

Comment:
There are already some images under `build-support/docker`... the centos 6 image is used on the binary-builder shards.

It would be good to move adding infrastructure into a separate PR though.

Reply:
Yeah, I was thinking of repackaging this PR in a least two PRs. One with the test fixes. 

The docker work ended up being less useful than I had hoped. Happy to see if anything I have can be rolled in to existing work in a new PR.